polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -789,11 +789,54 @@ class IDTypeTag {
   llvm::Regex regex;
 };
 
+template <typename TypeTag>
+class RaiseArrayCopyConstructorBasePattern
```

Comment:
This is very similar to `RaiseIDDefaultConstructor `, do you think there is anything that can be common?